bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/MetaDataKey.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.items;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * This class represents the key of a {@link MetaData} entity.
+ * It is a simple combination of a namespace and an item name.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+@NonNullByDefault
+public class MetaDataKey {
```

Comment:
Hm, this would be a good candidate for the cleaned up (Generic)UID base class usage. :wink:

Reply:
Nice - didn't have it on my radar either...